libhb/encavcodec.c

```diff
@@ -45,6 +45,9 @@ struct hb_work_private_s
     } frame_info[FRAME_INFO_SIZE];
 
     hb_chapter_queue_t * chapter_queue;
+
+    struct SwsContext  * sws_context_to_nv12;
```

Comment:
@galad87 Do your recent zimg additions have any bearing on the sws/nv12 conversion in this PR?

Reply:
I think so, AV_PIX_FMT_YUV420P shouldn't be hardcoded, it should check the job pixel format, and enable the conversion only if the job pixel format is not nv12.
We can reuse the same code for videotoolbox 10bit, so a more generic name would be better.

Reply:
I resolved most of this in recent commits and will let you improve with support for VT after merging.